### DISCUSSION TITLE
Allow nameof for unbound generic types

### DISCUSSION THREAD

**Discussion by fabianoliver** [dotnet/csharplang#5196] [Language Ideas] [answered]
It'd be quite nice if the following was possible:

```
public interface ISomething<T1,T2>
{
    T1 Hello(T2 world);
}

var nameOfHello = nameof(ISomething<,>.Hello); // Doesn't work
```

Now of course, you could do something like nameof(ISomething<object,object>.Hello. But aside from (presumably?) creating a type that isn't really needed at runtime, this can get quite tricky if there are intricate constraints on the generic parameters.

Being able to use nameof on unbound generic types could be quite useful for reflection, e.g. when needing to invoke methods on generic types created/bound at runtime.

**Comment by Joe4evr** [comment, marked as answer]
Duplicate of #702
